### PR TITLE
HPCC-19992 Add ability to protect files

### PR DESCRIPTION
**Pull request by miguelvazq**
Fixes HPCC-19992

Signed-off-by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: esp/src/src/ESPLogicalFile.ts [hunk 1]
```diff
@@ -237,15 +237,16 @@ var LogicalFile = declare([ESPUtil.Singleton], {    // jshint ignore:line
...
         //WsDfu/DFUInfo?FileName=progguide%3A%3Aexampledata%3A%3Akeys%3A%3Apeople.lastname.firstname&UpdateDescription=true&FileDesc=%C2%A0123&Save+Description=Save+Description
         var context = this;
         WsDfu.DFUInfo({
             request: {
                 Name: this.Name,
                 Cluster: this.Cluster,
                 UpdateDescription: true,
-                FileDesc: description
+                FileDesc: request.Description,
+                Protect: request.isProtected === true ? 1 : 2
```
Comment (GordonSmith) [inline esp/src/src/ESPLogicalFile.ts]: @wangkx - is this 1 || 2 correct?

**Comment by GordonSmith** [comment]
One sanity question for @wangkx otherwise code looks good.

File: esp/src/src/ESPLogicalFile.ts (hunk 1 above)
Comment (wangkx) [reply esp/src/src/ESPLogicalFile.ts]: 1 for Protect, 2 for Unprotect, 0 for NoChange

**Comment by richardkchapman** [comment]
@miguelvazq What version are you intending to target?

**Comment by miguelvazq** [comment]
7.2.0 as this was a new feature request

**Comment by miguelvazq** [comment]
@richardkchapman changed base branch to 7.2.x Should be ready to go.

**Pull request merged**